keps/sig-cloud-provider/2699-add-webhook-hosting-to-ccm/README.md

```diff
@@ -0,0 +1,455 @@
+This needs to be completed for cloud provider extraction to complete. Several
+Cloud Providers have indicated that this should be done inline, especially as the
+existing deprecated solution is an inline solution.
+
+### Goals
+
+Our current in tree Cloud Providers are able to stop using the existing PVL admission 
+controller and do so using the framework.
+
+### Non-Goals
+
+We are not intending to create a general admission webhook solution. This is just
+intended to host Cloud Provider specific webhooks as part of the Control Plane.
+
+## Proposal
+
+We will start by adding extension hooks which can be registered in the 
+cmd/cloud-controller-manager/main.go. This would be similar to the mechanism
+we already use to register new controllers. The existing sample shows this with
+a sample of registering the nodeipamcontroller which is not a normally installed
+controller in the cloud controller manager. In a similar way we will have a 
+sample of integrating a PVL mutating webhook into the sample CCM. We will also 
+have the system automatically detect if there are both controllers and webhooks
+registered in the binary. If both are registered it will automatically add 
+command line flags allowing webhooks and controller to be disabled. The 
+controller flag will default to the controllers being enabled. The webhook flag
+will default to the webhooks being disabled. We would also like to provide a 
+builder pattern for registering both the controller and webhook extensions.
+
+### User Stories (Optional)
+
+The users of this KEP are Cloud Providers. The intent is to make it easy for them
+to maintain the CCM controllers and webhooks across multiple versions. At the same
+time we are attempting to make it easy for the SIGs to make controllers or webhooks
+which can do what they know needs to be done and integrated into Cloud Provider 
+specific processes. We would like to do that in a way which makes merging upgrades
+relatively painless.
+
+#### Story 1
+
+Some Cloud Providers would prefer to keep controllers and webhooks in different 
+processses. They have concerns about attempting to run batch controllers in the 
+same process as webhooks which are "inline" and time sensitive. For these users
+it is easy to either build two different binaries or have the same binary act
+as two different binaries based on command line flags.
+
+#### Story 2
+
+For Cloud Providers who would like to keep things simple, it is easy to create
+a single processs which handles both controllers and webhooks.
+
+### Notes/Constraints/Caveats (Optional)
+
+
+### Risks and Mitigations
+
+Potentially there could be problems running webhooks and controller in the same
+process. Delays of 10 seconds or more can cause webhooks to fail. It is important 
+to understand than irrespective of failure mode on the webhook coniguration, 
+timeouts will always turn a webhook call into a FAIL. As such we are making it
```

Comment:
Are we also expecting to have the PVL webhook's policy be FAIL? I realize this KEP isn't focused on the PVL webhook but it might be cover the decision so we get ample design review.

Reply:
Leaving it as configurable but default to IGNORE and added a warning against the usage of FAIL.